Makefile

```diff
@@ -42,125 +26,85 @@ ifdef HTTPS_PROXY
 	DOCKERARGS += --build-arg https_proxy=$(HTTPS_PROXY)
 endif
 
-LDFLAGS=
+GO_BUILD_OPTS ?=
+GO_LDFLAGS ?=
+GO_FLAGS ?=
+
 ifdef STATIC
-	export CGO_ENABLED=0
-	LDFLAGS=-a -ldflags '-extldflags \"-static\"'
+	GO_BUILD_OPTS+= CGO_ENABLED=0
+	GO_LDFLAGS+= -extldflags \"-static\"
+	GO_FLAGS+= -a
 endif
 
-export GOPATH
-export GOBIN
-
 V = 0
 Q = $(if $(filter 1,$V),,@)
 
 .PHONY: all
-all: fmt lint build
+all: lint build test
 
-$(BASE): ; $(info  Setting GOPATH...)
-	@mkdir -p $(dir $@)
-	@ln -sf $(CURDIR) $@
+$(BINDIR): ; $(info Creating bin directory...)
+	$Q mkdir -p $@
 
-$(GOBIN):
-	@mkdir -p $@
+$(BUILDDIR): ; $(info Creating build directory...)
+	$Q mkdir -p $@
 
-$(BUILDDIR): | $(BASE) ; $(info Creating build directory...)
-	@cd $(BASE) && mkdir -p $@
+$(COVERAGE_DIR): ; $(info Creating coverage directory...)
+	$Q mkdir -p $@
 
-build: $(BUILDDIR)/$(BINARY_NAME) | ; $(info Building $(BINARY_NAME)...) @ ## Build SR-IOV Network device plugin
+.PHONY: build
+build: | $(BUILDDIR) ; $(info Building $(BINARY_NAME)...) @ ## Build SR-IOV Network device plugin
+	$Q cd $(CURDIR)/cmd/$(BINARY_NAME) && $(GO_BUILD_OPTS) go build -ldflags '$(GO_LDFLAGS)' $(GO_FLAGS) -o $(BUILDDIR)/$(BINARY_NAME) -tags no_openssl -v
 	$(info Done!)
 
-$(BUILDDIR)/$(BINARY_NAME): $(GOFILES) | $(BUILDDIR)
-	@cd $(BASE)/cmd/$(BINARY_NAME) && CGO_ENABLED=0 go build $(LDFLAGS) -o $(BUILDDIR)/$(BINARY_NAME) -tags no_openssl -v
-
-$(GOLINT): | $(BASE) ; $(info  building golint...)
-	$(call go-install-tool,$(GOLINT),golang.org/x/lint/golint@latest)
-
-$(GOCOVMERGE): | $(BASE) ; $(info  building gocovmerge...)
-	$(call go-install-tool,$(GOCOVMERGE),github.com/wadey/gocovmerge@latest)
+GOLANGCI_LINT = $(BINDIR)/golangci-lint
+GOLANGCI_LINT_VERSION ?= v1.55.2
```

Comment:
should we keep variables at the top of Makefile for ease?

Reply:
my preference is to keep those next to the relevant target thats downloading it :) it feels clearer to me.
a change in fashion (as in the past i was for keeping it at the top :D)

Reply:
Makes sense :)